### PR TITLE
~ adds project template samples

### PR DESCRIPTION
**Pull request by andreypudov**
# Description

Introduces project templates designed to help quickly create Data Parallel C++ application for FPGA and CPU/GPU targets.

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add an 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [X] Visual Studio
- [x] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [x] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [x] Implement coding guidelines and ensure code quality.
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [x] Adhere to readme template 
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Enforce format via the clang-format config file
- [x] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

File: DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/License.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2020, Intel Corporation. All rights reserved.
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/License.txt:1]: Please use this license
https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/README.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# "Hello World GPU" sample
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/README.md:1]: Old styleReadme, please update using https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot as a example

File: DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/License.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Copyright (c) 2020, Intel Corporation. All rights reserved.
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/License.txt:1]: Please use this license
https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: DirectProgramming/DPC++/ProjectTemplates/cmake-gpu/License.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+Copyright (c) 2020, Intel Corporation. All rights reserved.
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-gpu/License.txt:1]: Please use this license
https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: DirectProgramming/DPC++/ProjectTemplates/makefile-fpga/License.txt [hunk 5]
```diff
@@ -0,0 +1,7 @@
+Copyright (c) 2020, Intel Corporation. All rights reserved.
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/makefile-fpga/License.txt:1]: Please use this license
https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: DirectProgramming/DPC++/ProjectTemplates/makefile-gpu/License.txt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Copyright (c) 2020, Intel Corporation. All rights reserved.
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/makefile-gpu/License.txt:1]: Please use this license
https://github.com/oneapi-src/oneAPI-samples/blob/master/License.txt

File: DirectProgramming/DPC++/ProjectTemplates/makefile-gpu/README.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# Make based GPU Project Template
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/makefile-gpu/README.md:1]: Old styleReadme, please update using https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot as a template

File: DirectProgramming/DPC++/ProjectTemplates/makefile-fpga/README.md [hunk 8]
```diff
@@ -0,0 +1,65 @@
+# Make based FPGA Project Template
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/makefile-fpga/README.md:1]: Old styleReadme, please update using https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot as a template

File: DirectProgramming/DPC++/ProjectTemplates/cmake-gpu/README.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
+# CMake based GPU Project Template
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-gpu/README.md:1]: Old styleReadme, please update using https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot as a template

File: DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/README.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
+# CMake based FPGA Project Template
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/README.md:1]: Old styleReadme, please update using https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot as a template

File: DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/README.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+If GPU is not available on your system, you can fallback to cpu or default device.
+
+| Optimized for                     | Description
+|:---                               |:---
+| OS                                | Windows 10
+| Hardware                          | Integrated Graphics from Intel (GPU) GEN9 or higher
+| Software                          | Intel(R) oneAPI DPC++ Compiler (beta)
+| What you will learn               | Get started with DPC++ for GPU projects
+| Time to complete                  | n/a
+
```
Comment (JoeOster) [inline DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/README.md:14]: This is still an older style of readme, one example is there needs to be a purpose section to the readme, please take a look at https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC%2B%2B/CombinationalLogic/mandelbrot as an example

This comment is valid for all the readmes in this PR

File: DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/sample.json [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+          "cd build",
+          "cmake ..",
+          "make fpga_emu",
+          "make run_emu"
+        ]
+      },
+      {
+        "id": "report",
+        "steps": [
+          "cd build",
```
Comment (mkitez) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/sample.json:26]: `build` folder does not exist at this point, it has to be created. Each test object in the list is executed in a fresh directory, so the `build` folder is not retained from the previous build.

File: DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/README.md (hunk 11 above)
Comment (andreypudov) [reply DirectProgramming/DPC++/ProjectTemplates/Hello_World_GPU/README.md:14]: @JoeOster do you have requirements for the new style of readme files? What sections in your example are mondatory?

File: DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/sample.json [hunk 13]
```diff
@@ -23,6 +23,7 @@
       {
         "id": "report",
         "steps": [
+          "mkdir build",
           "cd build",
```
Comment (mkitez) [inline DirectProgramming/DPC++/ProjectTemplates/cmake-fpga/sample.json:27]: I guess you missed the `cmake ..` step

**Review by JoeOster — DISMISSED** [review]

**Review by JoeOster — APPROVED** [review]

**Review by anjgola — APPROVED** [review]

**Pull request merged**